tango/integrations/torch/train.py

```diff
@@ -466,12 +487,12 @@ def is_best_checkpoint() -> bool:
         A closure that we'll call when saving checkpoints to check if we should hardlink
         the best checkpoint path to the current checkpoint file.
         """
-        if val_metric is not None and best_val_metric is not None:
-            return (config.minimize_val_metric and val_metric <= best_val_metric) or (
-                not config.minimize_val_metric and val_metric >= best_val_metric
+        if val_metric is not None and best_val_metric_checkpointed is not None:
+            return (config.minimize_val_metric and val_metric <= best_val_metric_checkpointed) or (
+                not config.minimize_val_metric and val_metric >= best_val_metric_checkpointed
             )
         elif best_batch_loss is not None:
-            return best_batch_loss <= batch_loss
+            return batch_loss <= best_batch_loss
```

Comment:
I'm not sure we should keep the logic that keeps the best checkpoint based on batch loss. Batch loss is so noisy this is essentially random.

Reply:
Maybe "total loss since the last time we checkpointed". That would be less random. But that feels like a whole new feature.

Reply:
If we need some always-available fallback metric for model quality, I think recency will be better than "loss on the last batch before we checkpointed".

Reply:
Hmm yea using recency is probably okay. bfd336c

Reply:
I don't think you pushed that.

Reply:
Oops, yea push failed. Just pushed now